packages/components/src/Notification/ToastNotification/subcomponents/ToastNotificationManager/ToastNotificationManager.tsx

```diff
@@ -0,0 +1,139 @@
+import React from "react"
+import { createRoot } from "react-dom/client"
+import { v4 } from "uuid"
+import {
+  AddToastNotification,
+  ClearToastNotifications,
+  RemoveToastNotification,
+  ToastNotification,
+  ToastNotificationWithOptionals,
+} from "../../types"
+
+import { ToastNotificationsListContainer } from "../ToastNotificationsListContainer"
+
+type ToastNotificationApi = {
+  addToastNotification: AddToastNotification
+  removeToastNotification: RemoveToastNotification
+  clearToastNotifications: ClearToastNotifications
+}
+
+/**
+ * Persistent reference to a "portal" for rendering the separate DOM tree
+ */
+let portal: HTMLDivElement | undefined
+
+/**
+ * Set default values for optional params
+ */
+const setDefaults = (
+  notificationMaybe: ToastNotificationWithOptionals
+): ToastNotification => {
+  const copy = { ...notificationMaybe }
+  if (typeof copy.id === "undefined") {
+    copy.id = v4()
+  }
```

Comment:
Should swap this out for `useId()`

```
const reactId = useId()

if(...) { copy.id = reactId }
```

Reply:
Note: Sounds good, might not work. `useId` is a hook, which must be called within a component. This file is not comprised of components, it's just a bunch of functions (this one in particular is just a util for setting defaults) :( Worth trying though.

Reply:
Ah right, seems like you can't even render a Toast on the server side anyway? Ignore the comment lol

Reply:
Interesting 🤔 If I try to lift and shift this with `useId` its seems to break the component or at least the functions that are exported (ie: `addToastNotification`). While the component still renders when using the component version of `ToastNotification` all the function `addToastNotification` etc fail:

`Invalid hook call. Hooks can only be called inside of the body of a function component`

Starting to feel like we need to refactor the ToastNotifcationManager for this to work 🤔 Happy to hold on this to discuss at work review

Reply:
wüps, didn't see these comments before my schpeel 😆 I'll move these thoughts over to a ticket 😉